repos/system_upgrade/el7toel8/actors/checkskipphase/actor.py

```diff
@@ -0,0 +1,31 @@
+from leapp.actors import Actor
+from leapp.messaging.commands import SkipPhasesUntilCommand
+from leapp.reporting import Report
+from leapp.tags import ChecksPhaseTag, IPUWorkflowTag
+
+class CheckSkipPhase(Actor):
+    """
+    Skip following phases untill the report phase.
+
+    Following phases works with target user space - stuff around preparing and
+    checking the rpm transaction. We do not want to process those phases in case
+    of inhibition - e.g. for aspecific HW unsupported by the target system we
+    cannot do anything - we can just see some unclear errors in such case. So
+    we want to instead skip to the report phase to provide clear report to user
+    withou confusing errors.
+
+    The actor is processed as the last actor in the checks phase and it is the
```

Comment:
How does this work? What decides that this actor will really be executed last?

Reply:
It's based on leapp framework create execution order based on consumed messages. As the actor consumes the `Report` messages (and do not produce any), it will be resolved as the last one (otherwise I'd call it bug of leapp)